Sources/Core/Presenter/Channel/ChannelPresenter+Events.swift

```diff
@@ -28,15 +28,13 @@ extension ChannelPresenter {
         
         switch event {
         case .typingStart(let user, _, _):
-            if isThread {
+            guard isTypingEventsEnabled else {
                 return .none
             }
             
             let shouldUpdate = filterInvalidatedTypingUsers()
             
-            if channel.config.typingEventsEnabled,
-                !user.isCurrent,
-                (typingUsers.isEmpty || !typingUsers.contains(.init(user: user))) {
+            if !user.isCurrent, (typingUsers.isEmpty || !typingUsers.contains(.init(user: user))) {
```

Comment:
This `if` statement is hard to parse "visually". It would be nice to introduce some local variables to make it easier for your teammates to understand.

Reply:
I agree. But here, is it unclear for you?